crates/viewer/re_viewport_blueprint/src/entity_add_info.rs

```diff
@@ -0,0 +1,121 @@
+//! Utilities for determining if an entity can be added to a view.
+
+use nohash_hasher::IntMap;
+
+use re_entity_db::EntityTree;
+use re_log_types::EntityPath;
+use re_viewer_context::{DataQueryResult, ViewClassExt as _, ViewerContext};
+
+use crate::ViewBlueprint;
+
+/// Describes if an entity path can be added to a view.
+#[derive(Clone, PartialEq, Eq)]
+pub enum CanAddToView {
+    Compatible { already_added: bool },
+    No { reason: String },
+}
+
+impl Default for CanAddToView {
+    fn default() -> Self {
+        Self::Compatible {
+            already_added: false,
+        }
+    }
+}
+
+impl CanAddToView {
+    /// Can be generally added but view might already have this element.
+    pub fn is_compatible(&self) -> bool {
+        match self {
+            Self::Compatible { .. } => true,
+            Self::No { .. } => false,
+        }
+    }
+
+    /// Can be added and view doesn't have it already.
+    pub fn is_compatible_and_missing(&self) -> bool {
+        self == &Self::Compatible {
+            already_added: false,
+        }
+    }
+
+    pub fn join(&self, other: &Self) -> Self {
+        match self {
+            Self::Compatible { already_added } => {
+                let already_added = if let Self::Compatible {
+                    already_added: already_added_other,
+                } = other
+                {
+                    *already_added && *already_added_other
```

Comment:
isn't that on or? If it has been added to any of the two already it has been added to the union of both already?

Reply:
The semantics of "already added" is that you can no longer add such thing, so it's constraining. So you want a set of entities to be "already added" (aka you can no longer add it) _only_ if _all_ entity are already added.

Regardless, that was pure copy-paste, I've not even looked at the code 🤷🏻

Reply:
okay whatever :)